File: app/src/main/java/cat/xojan/random1/ui/adapter/HomePagerAdapter.java
package cat.xojan.random1.ui.adapter;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.ArrayList;
import java.util.List;

import cat.xojan.random1.R;

public class HomePagerAdapter extends FragmentPagerAdapter {

    private final List<Fragment> mFragmentList = new ArrayList<>();
    private final Context mContext;

    public HomePagerAdapter(FragmentManager fm, Context context) {
        super(fm);
        mContext = context;
    }

    @Override
    public Fragment getItem(int position) {
        return mFragmentList.get(position);
    }

    @Override
    public int getCount() {
        return mFragmentList.size();
    }

    public void addFragment(Fragment fragment) {
        mFragmentList.add(fragment);
    }

    @Override
    public CharSequence getPageTitle(int position) {
        switch (position) {
            case 0:
                return mContext.getString(R.string.podcasts_programs);
            case 1:
                return mContext.getString(R.string.podcasts_latest);
            case 2:
                return mContext.getString(R.string.podcasts_downloaded);
            default:
                throw new IllegalArgumentException("Invalid position: " + position);
        }
    }
}

File: app/src/main/java/cat/xojan/random1/ui/home/HomePagerAdapter.kt
package cat.xojan.random1.ui.home

import android.content.Context
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import cat.xojan.random1.R

class HomePagerAdapter(fm: FragmentManager,
                       private val context: Context) : FragmentPagerAdapter(fm) {

    private val fragmentList = mutableListOf<Fragment>()

    fun addFragment(fragment: Fragment) {
        fragmentList.add(fragment)
    }

    override fun getItem(position: Int): Fragment {
        return fragmentList[position]
    }

    override fun getCount(): Int {
        return fragmentList.size
    }

    override fun getPageTitle(position: Int): CharSequence = when (position) {
        0 -> context.getString(R.string.podcasts_programs)
        1 -> context.getString(R.string.podcasts_downloaded)
        else -> throw IllegalArgumentException("Invalid position: " + position)
    }
}